src/engine/strat_engine/cmd.rs

```diff
@@ -468,16 +468,22 @@ pub fn thin_metadata_size(
         })?
         .read_to_string(&mut output)?;
     if is_ok {
-        Ok(min(
-            THIN_META_MULT_FACTOR
+        let round = block_size - Sectors(1);
+        let determined_size = Sectors(
+            *(THIN_META_MULT_FACTOR
                 * Sectors(
                     output
                         .trim()
                         .parse::<u64>()
                         .map_err(|e| StratisError::Msg(e.to_string()))?,
-                ),
-            MAX_META_SIZE.sectors(),
-        ))
+                )
+                + round)
+                & !*round,
+        );
+        assert!(determined_size % block_size == Sectors(0));
+        let max = Sectors(*(MAX_META_SIZE.sectors() - round) & !*round);
```

Comment:
I don't understand why here ```round``` is subtracted from the value. I would have though adding it would be what was required.

Reply:
This is rounding down because if `MAX_META_SIZE` is not aligned to 1MiB, we want the largest multiple of 1 MiB that is below `MAX_META_SIZE`. After all, we don't want to round up to the nearest multiple if that is indeed the absolute max. Increasing it could cause an error.

Reply:
Ah, I think I see. You're recommending we just do the bitwise and at the end? I think you're correct about that.

Reply:
Resolved.